### PR TITLE
Wells from wellmatcher are sorted by insert order

### PR DESCRIPTION
**Pull request by joakim-hove**
Step on #2176

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/Schedule/Well/WellMatcher.cpp [hunk 1]
```diff
@@ -42,25 +57,25 @@ std::vector<std::string> WellMatcher::wells(const std::string& pattern) const {
...
             int flags = 0;
             if (fnmatch(pattern.c_str(), wname.c_str(), flags) == 0)
                 names.push_back(wname);
         }
-        return names;
+        return this->sort_wells( std::move(names) );
     }
 
-    auto name_iter = std::find(this->m_wells.begin(), this->m_wells.end(), pattern);
+    auto name_iter = std::find_if(this->m_wells.begin(), this->m_wells.end(), [&pattern](const std::pair<std::string, std::size_t>& arg) { return arg.first == pattern; });
```
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/Schedule/Well/WellMatcher.cpp:75]: I'm probably missing something.  `WellMatcher::m_wells` is an `unordered_map<string, size_t>` so why does this not use
```C++
auto name_iter = this->m_wells.find(pattern);
```
instead?

Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Schedule/Well/WellMatcher.cpp:75]: Ehhh - you are right ;-)

**Comment by joakim-hove** [comment]
This is not ready yet ...

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Pull request merged**